enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,272 @@
+But update requests like `desiredUpdate` have different schema requirements than output like `desired` and `availableUpdates`.
+For example, [the `force` property][api-force] makes sense in the `desiredUpdate` context, but not in the `availableUpdates` contexts.
+And release metadata like landing page URIs and available channels makes sense in the `desired` and `availableUpdates` contexts, but not in the `desiredUpdate` context.
+This enhancement adds a new `Release` type to be used in `status` properties so we can distinguish between input and output schema.
+
+## Motivation
+
+Cincinnati provides [metadata about releases][cincinnati-metadata], including (since [here][cincinnati-channel-metadata]) the channels to which the release currently belongs.
+Exposing that information to in-cluster consumers allows them to use live metadata, instead of talking to Cincinnati directly or hard-coding expected values.
+For example, the web console currently [hard-codes an expected set of available channels][console-channels] for a given release; with this enhancement they could look up available channels in `status.desired.additionalMetadata["io.openshift.upgrades.graph.release.channels"]`.
+
+### Goals
+
+* Exposing release metadata provided by upstream Cincinnati for the currently-desired and available-update releases.
+* Documenting some well-known metadata keys.
+
+### Non-Goals
+
+* Restricting metadata keys to those documented in this enhancement.
+    Cincinnati may continue to add and remove keys as they see fit as long as they provide at least the keys required by this enhancement.
+* Teaching Cincinnati to support requests without `channel` query parameters.
+    This might be useful for recovering from users specifying [`desiredUpdate`][api-desiredUpdate] that were not in their configured [`channel`][api-channel], but I am fine leaving recovery to the users.
+    They should have received sufficient information to adjust their channel from wherever they received the target release they are supplying.
+
+## Proposal
+
+As proposed in [this API pull request][api-pull-request], this enhancement adds a new type:
+
+```go
+// Release represents an OpenShift release image and associated metadata.
+// +k8s:deepcopy-gen=true
+type Release struct {
+  // version is a semantic versioning identifying the update version. When this
+  // field is part of spec, version is optional if image is specified.
+  Version string `json:"version"`
+
+  // image is a container image location that contains the update. When this
+  // field is part of spec, image is optional if version is specified and the
+  // availableUpdates field contains a matching version.
+  Image string `json:"image"`
+
+  // Metadata is metadata associated with the release, such
```

Comment:
I don't think this should be generic.  This should be typed.  It's either part of our API or not.  So `channels`, not `metadata.channels`.

Reply:
You should describe the mapping between metadata in the payload and the field, but that's it.

Reply:
> This should be typed. It's either part of our API or not.

That leaves no space for alpha-level metadata.  For example, we currently have a `description` property (e.g. see [here][1]), although we haven't put data in there since the 4.1 RCs:

```console
$ curl -sH 'Accept:application/json' 'https://api.openshift.com/api/upgrades_info/v1/graph?channel=prerelease-4.1&arch=amd64' | jq -r '.nodes[] | .version + ": " + .metadata.description' | sort -V | grep ': .'
4.1.0-rc.0: Beta 4
4.1.0-rc.3: beta 5
4.1.0-rc.4: Beta 6
4.1.0-rc.5: Beta 7
4.1.0-rc.6: Beta 8
```

If we decided some in-cluster consumer wanted access to that property without committing to supporting it forever, would that be an enhancement with an `alphaV1Description` property that we evaluate and eventually promote to a `description` property (or drop, if the evaluation decision was "we don't need this after all")?  If a customer for whatever reason wants to add additional metadata in their policy engine, do we want to stand in the way of that information being available in-cluster?  I'm not committed my unstructured, pass-through approach here, but I'd like more clarity around how we expect to handle evolution going forward if we pivot to a structured approach.

[1]: https://github.com/openshift/cincinnati/blob/cf6ab43f2fd10dba697b833b7f192a1f2e99b826/docs/design/cincinnati.md#example-1

Reply:
We handle alpha fields consistently, and alpha annotations has always been a tire fire.  Fields are always better than unstructured for use cases.  Don’t be generic.

Reply:
ok, rebased onto master and rerolled to use `Channels` and `Homepage` with e7ab1e7 -> 120eba5.  Please take a look :)